Library/Homebrew/test/test_integration_cmds.rb

```diff
@@ -42,8 +42,9 @@ def cmd_output(*args)
     cmd_args << "-rintegration_mocks"
     cmd_args << (HOMEBREW_LIBRARY_PATH/"../brew.rb").resolved_path.to_s
     cmd_args += args
+    cmd_exec = [RUBY_PATH] + cmd_args
     Bundler.with_original_env do
-      ENV["HOMEBREW_BREW_FILE"] = HOMEBREW_PREFIX/"bin/brew"
+      ENV["HOMEBREW_BREW_FILE"] = cmd_exec.join " "
```

Comment:
Yes, that's what `cmd_id_from_args` is for. The nested `brew readall` spawned from `brew tap` doesn't go through this code again (unless I'm missing something). Thus it re-uses the same ID and gets its result overwritten by the `brew tap` process that finishes later.

If this worked as it probably should, I'd expect `brew tests --coverage --only=integration_cmds/tap` to have nonzero coverage for `cmd/readall.rb`, but that's not something I can observe locally.